Error message:
```
/usr/local/bin/bazel
2022/[11](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:12)/21 23:15:52 Downloading https://releases.bazel.build/5.3.2/release/bazel-5.3.2-linux-x86_64...
bazel 5.3.2
Extracting Bazel installation...
DEBUG: Rule 'com_google_protobuf' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "f66073dee0bc159157b0bd7f502d7d1ee0bc76b3c1eac9836927511bdc4b3fc1"
DEBUG: Repository com_google_protobuf instantiated at:
  /home/runner/work/gax-java/gax-java/WORKSPACE:13:37: in <toplevel>
  /home/runner/work/gax-java/gax-java/repositories.bzl:51:11: in com_google_api_gax_java_repositories
  /home/runner/work/gax-java/gax-java/repositories.bzl:139:14: in _maybe
Repository rule http_archive defined at:
  /home/runner/.cache/bazel/_bazel_runner/708235df2b9026c2ff59e4840c097026/external/bazel_tools/tools/build_defs/repo/http.bzl:355:31: in <toplevel>
DEBUG: Rule 'io_grpc_grpc_java' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "e258976ca0404fc6a6113a50add56857062828458f6[12](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:13)85c1b43f0bfae305e9e"
DEBUG: Repository io_grpc_grpc_java instantiated at:
  /home/runner/work/gax-java/gax-java/WORKSPACE:[13](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:14):37: in <toplevel>
  /home/runner/work/gax-java/gax-java/repositories.bzl:60:11: in com_google_api_gax_java_repositories
  /home/runner/work/gax-java/gax-java/repositories.bzl:139:[14](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:15): in _maybe
Repository rule http_archive defined at:
  /home/runner/.cache/bazel/_bazel_runner/708235df2b9026c2ff59e4840c097026/external/bazel_tools/tools/build_defs/repo/http.bzl:355:31: in <toplevel>
DEBUG: Rule 'bazel_skylib' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "28f81e36692e1d87823623a99966b2daf85af3fdc1b40f98e37bd5294f3dd185"
DEBUG: Repository bazel_skylib instantiated at:
  /home/runner/work/gax-java/gax-java/WORKSPACE:13:37: in <toplevel>
  /home/runner/work/gax-java/gax-java/repositories.bzl:68:11: in com_google_api_gax_java_repositories
  /home/runner/work/gax-java/gax-java/repositories.bzl:139:14: in _maybe
Repository rule http_archive defined at:
  /home/runner/.cache/bazel/_bazel_runner/708235df2b9026c2ff59e4840c097026/external/bazel_tools/tools/build_defs/repo/http.bzl:355:31: in <toplevel>
DEBUG: /home/runner/.cache/bazel/_bazel_runner/708235df2b9026c2ff59e4840c097026/external/io_grpc_grpc_java/java_grpc_library.bzl:76:14: in srcs attribute of @io_grpc_grpc_java//grpclb:load_balancer_java_grpc: Proto source with label @io_grpc_grpc_proto//:grpclb_load_balancer_proto should be in same package as consuming rule
INFO: Analyzed 133 targets (170 packages loaded, 2851 targets configured).
INFO: Found 11 targets and 122 test targets...
INFO: From Compiling src/google/protobuf/message_lite.cc:
In file included from /usr/include/string.h:535,
                 from external/com_google_protobuf/src/google/protobuf/stubs/port.h:39,
                 from external/com_google_protobuf/src/google/protobuf/stubs/common.h:48,
                 from external/com_google_protobuf/src/google/protobuf/message_lite.h:46,
                 from external/com_google_protobuf/src/google/protobuf/message_lite.cc:36:
In function 'void* memcpy(void*, const void*, size_t)',
    inlined from 'uint8_t* google::protobuf::io::EpsCopyOutputStream::WriteRaw(const void*, int, uint8_t*)' at external/com_google_protobuf/src/google/protobuf/io/coded_stream.h:684:16,
    inlined from 'virtual uint8_t* google::protobuf::internal::ImplicitWeakMessage::_InternalSerialize(uint8_t*, google::protobuf::io::EpsCopyOutputStream*) const' at external/com_google_protobuf/src/google/protobuf/implicit_weak_message.h:103:28,
    inlined from 'bool google::protobuf::MessageLite::SerializePartialToZeroCopyStream(google::protobuf::io::ZeroCopyOutputStream*) const' at external/com_google_protobuf/src/google/protobuf/message_lite.cc:411:30:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33: warning: 'void* __builtin___memcpy_chk(void*, const void*, long unsigned int, long unsigned int)' specified size between 1844674407[15](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:16)62067968 and 1844674407370955[16](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:17)15 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
INFO: From Compiling src/google/protobuf/message_lite.cc [for host]:
In file included from /usr/include/string.h:535,
                 from external/com_google_protobuf/src/google/protobuf/stubs/port.h:39,
                 from external/com_google_protobuf/src/google/protobuf/stubs/common.h:48,
                 from external/com_google_protobuf/src/google/protobuf/message_lite.h:46,
                 from external/com_google_protobuf/src/google/protobuf/message_lite.cc:36:
In function 'void* memcpy(void*, const void*, size_t)',
    inlined from 'uint8_t* google::protobuf::io::EpsCopyOutputStream::WriteRaw(const void*, int, uint8_t*)' at external/com_google_protobuf/src/google/protobuf/io/coded_stream.h:684:16,
    inlined from 'virtual uint8_t* google::protobuf::internal::ImplicitWeakMessage::_InternalSerialize(uint8_t*, google::protobuf::io::EpsCopyOutputStream*) const' at external/com_google_protobuf/src/google/protobuf/implicit_weak_message.h:103:28,
    inlined from 'bool google::protobuf::MessageLite::SerializePartialToZeroCopyStream(google::protobuf::io::ZeroCopyOutputStream*) const' at external/com_google_protobuf/src/google/protobuf/message_lite.cc:411:30:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33: warning: 'void* __builtin___memcpy_chk(void*, const void*, long unsigned int, long unsigned int)' specified size between [18](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:19)446744071562067968 and 18446744073709551615 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
FAIL: //gax:com.google.api.gax.retrying.ScheduledRetryingExecutorTest (see /home/runner/.cache/bazel/_bazel_runner/708235df2b9026c2ff59e4840c097026/execroot/com_google_api_gax_java/bazel-out/k8-fastbuild/testlogs/gax/com.google.api.gax.retrying.ScheduledRetryingExecutorTest/test.log)
INFO: From Testing //gax:com.google.api.gax.retrying.ScheduledRetryingExecutorTest:
==================== Test output for //gax:com.google.api.gax.retrying.ScheduledRetryingExecutorTest:
JUnit4 Test Runner
..................E............
Time: 5.[19](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:20)8
There was 1 failure:
1) testCancelOuterFutureAfterStart[Custom retry settings: true](com.google.api.gax.retrying.ScheduledRetryingExecutorTest)
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at com.google.api.gax.retrying.ScheduledRetryingExecutorTest.testCancelOuterFutureAfterStart(ScheduledRetryingExecutorTest.java:269)

FAILURES!!!
Tests run: 30,  Failures: 1


BazelTestRunner exiting with a return value of 1
JVM shutdown hooks (if any) will run now.
The JVM will exit once they complete.

-- JVM shutdown starting at [20](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:21)22-11-[21](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:22) 23:[22](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:23):[46](https://github.com/googleapis/gax-java/actions/runs/3518888871/jobs/5898273210#step:8:47) --

================================================================================
```

--- a/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java
+++ b/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java
@@ -326,7 +326,9 @@ public final class InstantiatingGrpcChannelProvider implements TransportChannelP
           GoogleDefaultChannelCredentials.newBuilder().callCredentials(callCreds).build();
       isDirectPathXdsEnabled = Boolean.parseBoolean(envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS));
       if (isDirectPathXdsEnabled) {
-        // google-c2p resolver target must not have a port number
+        // google-c2p: CloudToProd(C2P) Directpath. This scheme is defined in
+        // io.grpc.googleapis.GoogleCloudToProdNameResolverProvider.
+        // This resolver target must not have a port number.
         builder = Grpc.newChannelBuilder("google-c2p:///" + serviceAddress, channelCreds);
       } else {
         builder = Grpc.newChannelBuilderForAddress(serviceAddress, port, channelCreds);

